library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -361,6 +405,16 @@ public Slice get(Slice key, long timeBucket, ReadSource readSource)
      */
     private BucketedValue getValueFromTimeBucketReader(Slice key, long timeBucket)
     {
+      if (bloomFilter != null) {
```

Comment:
Agreed. Makes sense.
Can we move the usage of bloom filter to the `getFromReaders()` call? That is the first place where we know that files might be accessed.